### ISSUE TITLE
Safety Button can't back to lock status even in disarm

### ISSUE THREAD

**Issue by bys1123** [PX4/PX4-Autopilot#17369]
After flash current master, Safety Button can't back to lock status

**Comment by saengphet** [comment]
> After flash current master, Safety Button can't back to lock status on Pixhawk4

Hi, @bys1123 I have tested with cube orange and got the same result. After press the safety switch button once to pre-arm on state, it could not turn to pre-arm off state, the LED still be "double blink" instead of slow blink.

**Comment by LorenzMeier** [comment]
This is by design (see the release notes) because we have seen accidental in-air triggering of the safety switch due to poor wiring. So now it is latching.

**Comment by bys1123** [comment]
@LorenzMeier Thanks for explain, but in disarm status, i think safety button is still useful, especially in research environment. In arm status, I agree make safety switch un-tigger-able.

**Comment by ThomasRigi** [comment]
@LorenzMeier Are there any plans to fix this for v1.12? I agree that latching is good for when the drone is armed, but it has be possible to un-prearm when the drone is disarmed as @bys1123 is saying.

**Comment by ThomasRigi** [comment]
And could you please send the reference to the PR that changed to latching behaviour? It's not linked in the release notes

**Comment by LorenzMeier** [comment]
I will look into it. My main concern is to keep it consistent and I need to see if that's consistently doable before merging the PX4 IO rework (which will definitely allow for it).

**Comment by dlech** [comment]
> And could you please send the reference to the PR that changed to latching behaviour? It's not linked in the release notes

I searched for related commits:
- https://github.com/PX4/PX4-Autopilot/commit/867b006861d4ea887d1f17a79d84b4455a392c91
- https://github.com/PX4/PX4-Autopilot/commit/7157584fe11dbcf644928f877964d0c2ff7dad90
- https://github.com/PX4/PX4-Autopilot/pull/16751

**Comment by spurnvoj** [comment]
Hi, any update on this?

**Comment by ThomasRigi** [comment]
@LorenzMeier @MaEtUgR @dagar can we get this fixed ?

**Comment by dagar** [comment]
> @LorenzMeier @MaEtUgR @dagar can we get this fixed ?

@ThomasRigi yes, maybe we can talk about it in https://github.com/PX4/PX4-Autopilot/pull/19558 and review your specific requirements?

The original change to have the px4io safety button latch was to remove the possibility of accidental disarm in air. Additionally we also changed the safety button to disabled by default because that's what the vast majority of people were doing and it doesn't really for so many systems that use more than just the px4io outputs.

Anyway, sorry if this disrupted anything for you, let's review the core requirements?

**Comment by ThomasRigi** [comment]
Ok, I commented in #19558. :+1:
